internal/controller/bucket_controller.go

```diff
@@ -196,55 +228,183 @@ func (r BucketReconciler) SetupWithManager(mgr ctrl.Manager) error {
 func (r BucketReconciler) getRoleArn(ctx context.Context) (string, error) {
 	logger := log.FromContext(ctx)
 
+	// cluster := &unstructured.Unstructured{}
```

Comment:
Can we create a clean abstraction liké somekind of cluster service instead of a provider Map though ?